### ISSUE TITLE
Unable to build timeline

### ISSUE THREAD

**Issue by jczacharia** [visjs/vis-timeline#59]
tl;dr: How do I properly build the timeline? It seems to be broken event with a fresh install.

---

I am experiencing issues trying to build the master branch.

This is from a fresh clone and after running `npm install`.

**1.** Running `npm run build` gives the error:
```
$ npm run build

> vis-timeline@5.1.0 build C:\Users\zacharjc\Desktop\Repos\vis-timeline
> rollup --config rollup.config.js


index.js → dist/vis-timeline-graph2d.esm.js...
created dist/vis-timeline-graph2d.esm.js in 8.8s

index.js → dist/vis-timeline-graph2d.min.js...
[!] (plugin babel-minify) Error: don't know how to turn this value into a node
Error: don't know how to turn this value into a node
    at Object.valueToNode (C:\Users\zacharjc\Desktop\Repos\vis-timeline\node_modules\@babel\types\lib\converters\valueToNode.js:103:9)
    at PluginPass.Expression (C:\Users\zacharjc\Desktop\Repos\vis-timeline\node_modules\babel-plugin-minify-constant-folding\lib\index.js:195:26)
    at PluginPass.newFn (C:\Users\zacharjc\Desktop\Repos\vis-timeline\node_modules\@babel\traverse\lib\visitors.js:230:17)
    at newFn (C:\Users\zacharjc\Desktop\Repos\vis-timeline\node_modules\@babel\traverse\lib\visitors.js:193:21)
    at NodePath._call (C:\Users\zacharjc\Desktop\Repos\vis-timeline\node_modules\@babel\traverse\lib\path\context.js:53:20)
    at NodePath.call (C:\Users\zacharjc\Desktop\Repos\vis-timeline\node_modules\@babel\traverse\lib\path\context.js:40:17)
    at NodePath.visit (C:\Users\zacharjc\Desktop\Repos\vis-timeline\node_modules\@babel\traverse\lib\path\context.js:88:12)
    at TraversalContext.visitQueue (C:\Users\zacharjc\Desktop\Repos\vis-timeline\node_modules\@babel\traverse\lib\context.js:118:16)
    at TraversalContext.visitSingle (C:\Users\zacharjc\Desktop\Repos\vis-timeline\node_modules\@babel\traverse\lib\context.js:90:19)
    at TraversalContext.visit (C:\Users\zacharjc\Desktop\Repos\vis-timeline\node_modules\@babel\traverse\lib\context.js:146:19)
```

The workaround is to just use the `vis-timeline-graph2d.esm.js` file

---

 **2.** In my project, I have installed the local vis-timeline vis `npm install -s ../vis-timeline` which renders:
```
"dependencies": {
    ...
    "vis-timeline": "file:../vis-timeline"
    ...
}
```
in package.json.

When importing vis-timeline via `import * as vis from 'vis-timeline';` in my typescript project using this code:
```
this.timeline = new vis.Timeline(
      document.getElementById('timeline-main'),
      [
        { id: 1, content: 'item 1', start: '2014-04-20' },
        { id: 2, content: 'item 2', start: '2014-04-14' },
        { id: 3, content: 'item 3', start: '2014-04-18' },
        { id: 4, content: 'item 4', start: '2014-04-16', end: '2014-04-19' },
        { id: 5, content: 'item 5', start: '2014-04-25' },
        { id: 6, content: 'item 6', start: '2014-04-27', type: 'point' }
      ],
      { height: 'calc(100vh - 136px)' }
    );
```
I get the error:
```
"export 'Timeline' (imported as 'vis') was not found in 'vis-timeline'
```

But when importing vis `const vis = require('vis-timeline');`

I get the error:
```
ERROR TypeError: vis.Timeline is not a constructor
    at ActivitiesComponent.push../src/app/features/crew/component/activities/activities.component.ts.ActivitiesComponent.ngAfterContentInit (activities.component.ts:19)
    at callProviderLifecycles (core.js:21406)
    at callElementProvidersLifecycles (core.js:21387)
    at callLifecycleHooksChildrenFirst (core.js:21377)
    at checkAndUpdateView (core.js:29442)
    at callViewAction (core.js:29679)
    at execComponentViewsAction (core.js:29621)
    at checkAndUpdateView (core.js:29444)
    at callViewAction (core.js:29679)
    at execEmbeddedViewsAction (core.js:29642)
```

I want to build the timeline so I can contribute and test out the timeline-plus merges but I am blocked from doing so until I can do a successful build.

**Comment by jczacharia** [comment]
I did a fresh install of master. The `Error: don't know how to turn this value into a node` is now gone (not sure why). But I am still getting "export 'Timeline' (imported as 'vis') was not found in 'vis-timeline'`

Are dependencies being exported correctly?

**Comment by jczacharia** [comment]
Solved it. If you are using angular/typescript then you need to declare: `declare var vis: any;` at the top of your file.
